### ISSUE TITLE
Crash when trying to view a file that appears to point to nonsensical sectors

### ISSUE THREAD

**Issue by bhtooefr** [fadden/ciderpress#42]
On [this disk image](https://www.apple.asimov.net/images/disk_utils/diagnostics/Apple%20II%2B%20Dealer%20Diagnostics.dsk) of the Apple II+ Dealer Diagnostics, attempting to open the file DIAGNOSTICS-16 SECTOR causes a crash after a couple of seconds:

```
Log Name:      Application
Source:        Application Error
Date:          2020-08-03 00:47:32
Event ID:      1000
Task Category: (100)
Level:         Error
Keywords:      Classic
User:          N/A
Computer:      minenwerfer
Description:
Faulting application name: CiderPress.exe, version: 4.0.0.0, time stamp: 0x5f1a05d8
Faulting module name: CiderPress.exe, version: 4.0.0.0, time stamp: 0x5f1a05d8
Exception code: 0xc0000005
Fault offset: 0x0004ce7a
Faulting process id: 0x36f4
Faulting application start time: 0x01d669512c3d9954
Faulting application path: C:\Program Files (x86)\faddenSoft\CiderPress\CiderPress.exe
Faulting module path: C:\Program Files (x86)\faddenSoft\CiderPress\CiderPress.exe
Report Id: 62a4601d-5867-4e99-8688-fbc7e0785e7c
Faulting package full name: 
Faulting package-relative application ID: 
Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Application Error" />
    <EventID Qualifiers="0">1000</EventID>
    <Level>2</Level>
    <Task>100</Task>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2020-08-03T04:47:32.190674600Z" />
    <EventRecordID>40251</EventRecordID>
    <Channel>Application</Channel>
    <Computer>minenwerfer</Computer>
    <Security />
  </System>
  <EventData>
    <Data>CiderPress.exe</Data>
    <Data>4.0.0.0</Data>
    <Data>5f1a05d8</Data>
    <Data>CiderPress.exe</Data>
    <Data>4.0.0.0</Data>
    <Data>5f1a05d8</Data>
    <Data>c0000005</Data>
    <Data>0004ce7a</Data>
    <Data>36f4</Data>
    <Data>01d669512c3d9954</Data>
    <Data>C:\Program Files (x86)\faddenSoft\CiderPress\CiderPress.exe</Data>
    <Data>C:\Program Files (x86)\faddenSoft\CiderPress\CiderPress.exe</Data>
    <Data>62a4601d-5867-4e99-8688-fbc7e0785e7c</Data>
    <Data>
    </Data>
    <Data>
    </Data>
  </EventData>
</Event>
```

I would fully expect an error message upon attempting to read the file - upon trying to view the file in Copy II Plus 8.4, I received an I/O error at track $E7, sector $A5, which is utterly impossible, so this image is either corrupt or not entirely what it says it is filesystem-wise:

![image](https://user-images.githubusercontent.com/1939057/89147051-9a72ae00-d523-11ea-9c22-fcf2ad068dfb.png)

However, CiderPress crashing indicates to me that there could be a buffer overflow, and that may have security implications.

**Comment by fadden** [comment]
I can reproduce it.  There's a null pointer dereference when it tries to reformat the file for viewing.  Something along the way missed an opportunity to report an error.  I'll look at it tomorrow.

Thanks for the report!

**Comment by fadden** [comment]
The "is it S-C assembler?" code was missing a test, so a file that had type=INT, auxType=0, and length=0 would cause a crash.  Fixed in 4.0.5-d1.
